pkg/virt-handler/backoff/backoff_test.go

```diff
@@ -0,0 +1,165 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021
+ *
+ */
+
+package backoff
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("limited backoff", func() {
+	const backoffDeadlineDuration = 20 * time.Millisecond
+	const initialBackoff = 3 * time.Second
+	const maxBackoff = 10 * time.Minute
+	const backoffFactor = 1.8
+	const backoffSteps = 10
+	const backoffJitter = 0.0
+
+	var testBackoff *limitedBackoff
+
+	BeforeEach(func() {
+		testBackoff = newLimitedBackoff(backoffDeadlineDuration, initialBackoff, maxBackoff, backoffSteps, backoffFactor, backoffJitter)
+	})
+
+	It("should be created with valid backoff and deadline time", func() {
+		Expect(testBackoff.initialBackoff).To(Equal(testBackoff.backoff))
+		Expect(time.Now().Before(testBackoff.deadline)).To(BeTrue())
+		Expect(time.Now().After(testBackoff.deadline)).To(BeFalse())
```

Comment:
The guideline is that the `Expect` argument is the actual result/value and then the expected one.
Here we have a mix/mess.

How about using `BeTemporally`? (see gomega matchers).

Reply:
Done